examples/graphiql-cdn/index.html

```diff
@@ -0,0 +1,81 @@
+<!--
+ *  Copyright (c) 2019 GraphQL Contributors
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the license found in the
+ *  LICENSE file in the root directory of this source tree.
+-->
+<!DOCTYPE html>
+<html>
+  <head>
+    <style>
+      body {
+        height: 100%;
+        margin: 0;
+        width: 100%;
+        overflow: hidden;
+      }
+      #graphiql {
+        height: 100vh;
+      }
+    </style>
+
+    <!--
+      This GraphiQL example depends on Promise and fetch, which are available in
+      modern browsers, but can be "polyfilled" for older browsers.
+      GraphiQL itself depends on React DOM.
+      If you do not want to rely on a CDN, you can host these files locally or
+      include them directly in your favored resource bunder.
+    -->
+    <script
+      crossorigin
+      src="https://unpkg.com/react@16/umd/react.development.js"
+    ></script>
+    <script
+      crossorigin
+      src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"
+    ></script>
+
+    <!--
+      These two files can be found in the npm module, however you may wish to
+      copy them directly into your environment, or perhaps include them in your
+      favored resource bundler.
+     -->
+    <link rel="stylesheet" href="https://unpkg.com/graphiql/graphiql.min.css" />
+  </head>
+  <body>
+    <div id="graphiql">Loading...</div>
+    <script
+      src="https://unpkg.com/graphiql/graphiql.min.js"
+      type="application/javascript"
+    ></script>
+    <script src="/renderExample.js" type="application/javascript"></script>
+    <script>
+      function graphQLFetcher(graphQLParams) {
+        return fetch('https://swapi-graphql.netlify.com/.netlify/functions/index', {
+          method: 'post',
+          headers: {
+            Accept: 'application/json',
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(graphQLParams),
+          credentials: 'include',
+        }).then(function(response) {
+          try {
+            return response.json();
```

Comment:
This'll only catch the error if it's thrown synchronously, which it's unlikely to be. Since we're not using `async/await` or arrow functions here ( :disappointed: ) a better solution is to extend the promise chain:

```
}).then(function(response) {
  return response.json().catch(function () { return response.text(); });
});
```

Reply:
Alternatively we could rewrite the entire thing using `async/await` and leave supporting older browsers as an exercise to the reader:

```js
async function graphQLFetcher(graphQLParams) {
  const response = await fetch(
    'https://swapi-graphql.netlify.com/.netlify/functions/index',
    {
      method: 'post',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(graphQLParams),
      credentials: 'same-origin',
    }
  );
  try {
    return await response.json();
  } catch (error) {
    return response.text();
  }
}
```

I've at least left in the unused catch binding to support _some_ browsers :wink:

Reply:
we could! since it's a CDN example i'm torn between whether we should aim for es5 compatibility or if only modern features should be used.

reason being is that CDN examples allow people building runtimes in other languages to not have to deal with javascript almost at all if they dont need to. i guess we could provide an ie11 compatible example? might need a few polyfills. could add a cypress sub-suite to do a regression suite on that eventually once cypress supports ie11. or add testcafe or something temporarily with browserstack/etc for IE confirmation. if we decide we want to natively support IE11, rather than just making some "suggestions" but not officially supporting it.

Reply:
but on the other side, if they do need to interact with the cdn example a bit more to add some basic jwt auth or something, then es6 would make the learning curve slightly easier.